pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -261,15 +293,17 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 			s.updateCachedPodData(pod)
 		}
 	}
-	UnfinishedWorkSeconds.Delete(map[string]string{ControllerLabel: injection.GetControllerName(ctx), schedulingIDLabel: string(s.id)})
+	UnfinishedWorkSeconds.Delete(map[string]string{ControllerLabel: injection.GetControllerName(ctx), schedulingIDLabel: string(s.uuid)})
 	for _, m := range s.newNodeClaims {
 		m.FinalizeScheduling()
 	}
 
 	return Results{
 		NewNodeClaims: s.newNodeClaims,
 		ExistingNodes: s.existingNodes,
-		PodErrors:     errors,
+		PodErrors: lo.OmitBy(errors, func(_ *corev1.Pod, err error) bool {
```

Comment:
What do you think -- should we omit these higher up when we actually print out the errors? Realistically, it does make sense to me that this is included in these pod errors (from what gets returned in the results section) because this is the reason that the pod didn't schedule -- I think the real bit is that we don't want to print it back

Reply:
I don't have a strong opinion either way, you're right that at the moment the main goal is just preventing us from printing those errors. We can move this filter further up the call stack. I think I originally handled it here so we wouldn't need to duplicate the check in the provisioner and the disruption controller, but we should only see these errors in provisioning anyway due to the fallback policy.